firefox-ios/firefox-ios-tests/Tests/ClientTests/ShareManagerTests.swift

```diff
@@ -212,9 +212,183 @@ final class ShareManagerTests: XCTestCase {
+        XCTAssertEqual(itemForActivity as? URL, testWebURL)
+
+        // The rest of the content should be unchanged from other tests:
+        _ = try XCTUnwrap(activityItems[safe: 1] as? TabPrintPageRenderer)
+
+        _ = try XCTUnwrap(activityItems[safe: 2] as? TabWebView)
+
+        let titleActivityItemProvider = try XCTUnwrap(activityItems[safe: 3] as? TitleActivityItemProvider)
+        XCTAssertEqual(
+            titleActivityItemProvider.item as? String,
+            testWebpageDisplayTitle,
+            "When no explicit share message is set, we expect to see the webpage's title."
+        )
+    }
+
+    func testGetActivityItems_forTab_withSentFromFirefoxDisabled_DoesNotOverride() throws {
+        setupNimbusSentFromFirefoxTesting(isEnabled: false, isTreatmentA: true)
+
+        let whatsAppActivityIdentifier = "net.whatsapp.WhatsApp.ShareExtension"
+        let whatsAppActivity = UIActivity.ActivityType(rawValue: whatsAppActivityIdentifier)
+
+        let activityItems = ShareManager.getActivityItems(
+            forShareType: .tab(url: testWebURL, tab: testTab),
+            withExplicitShareMessage: nil
+        )
+
+        // Check we get all 4 types of share items for tabs below
+        XCTAssertEqual(activityItems.count, 4)
+
+        let urlActivityItemProvider = try XCTUnwrap(activityItems[safe: 0] as? URLActivityItemProvider)
+        let urlDataIdentifier = urlActivityItemProvider.activityViewController(
+            createStubActivityViewController(),
+            dataTypeIdentifierForActivityType: whatsAppActivity
+        )
+        let itemForActivity = urlActivityItemProvider.activityViewController(
+            createStubActivityViewController(),
+            itemForActivityType: whatsAppActivity
+        )
+
+        XCTAssertEqual(urlDataIdentifier, UTType.url.identifier)
+        XCTAssertEqual(itemForActivity as? URL, testWebURL)
+
+        // The rest of the content should be unchanged from other tests:
+        _ = try XCTUnwrap(activityItems[safe: 1] as? TabPrintPageRenderer)
+
+        _ = try XCTUnwrap(activityItems[safe: 2] as? TabWebView)
+
+        let titleActivityItemProvider = try XCTUnwrap(activityItems[safe: 3] as? TitleActivityItemProvider)
+        XCTAssertEqual(
+            titleActivityItemProvider.item as? String,
+            testWebpageDisplayTitle,
+            "When no explicit share message is set, we expect to see the webpage's title."
+        )
+    }
+
     // MARK: - Helpers
 
     private func createStubActivityViewController() -> UIActivityViewController {
         return UIActivityViewController(activityItems: [], applicationActivities: [])
     }
+
+    private func setupNimbusSentFromFirefoxTesting(isEnabled: Bool, isTreatmentA: Bool) {
+        FxNimbus.shared.features.sentFromFirefoxFeature.with { _, _ in
+            return SentFromFirefoxFeature(
+                enabled: isEnabled,
+                isTreatmentA: isTreatmentA
+            )
+        }
+    }
```

Comment:
Heya @adudenamedruby, I followed you example to test Nimbus feature flags in our unit tests.

I was wondering, do we have an established way to inject different `.buildOnly` vs. `.userOnly` preferences? 🤔  I'd like to make sure we account for the user's preference overriding the nimbus setting (e.g. with a on/off toggle in general Settings). It's easy to accidentally choose `.buildOnly` in the client code when we should really be using `.buildAndUser`. Do you think I'd have to edit the UserDefaults via pref keys to accomplish this?

Reply:
Ooo, interesting. I haven't really had a case where I've had to test that actually, so my first thought is that, yes, we'd have to edit the UserDefaults stuff to test.

Reply:
Cool, thanks! I am going to make myself a ticket to dig into this a bit next week. FXIOS-10881